src/modules/flight_mode_manager/tasks/AutoFollowTarget/FlightTaskAutoFollowTarget.cpp

```diff
@@ -43,330 +43,386 @@
-		const Vector3f target_predicted_position = predict_future_pos_ned_est(POSITION_FILTER_ALPHA, pos_ned_est, vel_ned_est,
-				acc_ned_est);
-		_target_position_filter.update(target_predicted_position);
-	}
+		case kFollowAltitudeModeConstant:
 
-	return _target_position_filter.getState();
+		//FALLTHROUGH
 
+		default:
+			gimbal_height = _home_position_z - _position(2); // Assume target is at home position's altitude
+			break;
+	}
+
+	return gimbal_height;
 }
 
+
 bool FlightTaskAutoFollowTarget::update()
 {
-	_follow_target_estimator_sub.update(&_follow_target_estimator);
 	follow_target_status_s follow_target_status{};
+	bool in_emergency_ascent{false};
+	float gimbal_pitch{NAN};
+	float raw_orbit_angle_setpoint{NAN};
+
+	// Get the latest target estimator message for target position and velocity
+	_follow_target_estimator_sub.update(&_follow_target_estimator);
 
 	if (_follow_target_estimator.timestamp > 0 && _follow_target_estimator.valid) {
-		const Vector3f pos_ned_est{_follow_target_estimator.pos_est};
-		const Vector3f vel_ned_est{_follow_target_estimator.vel_est};
-		const Vector3f acc_ned_est{_follow_target_estimator.acc_est};
-		const Vector3f target_position_filtered = calculate_target_position_filtered(pos_ned_est, vel_ned_est, acc_ned_est);
-		const Vector2f offset_vector_filtered = calculate_offset_vector_filtered(vel_ned_est);
-		const Vector3f drone_desired_position = calculate_drone_desired_position(target_position_filtered,
-							offset_vector_filtered);
+		updateTargetPositionVelocityFilter(_follow_target_estimator);
+		const Vector3f target_position_filtered = _target_position_velocity_filter.getState();
+		const Vector3f target_velocity_filtered = _target_position_velocity_filter.getRate();
+		const Vector2f drone_to_target_vector  = Vector2f(target_position_filtered.xy()) - Vector2f(_position.xy());
+
+		// Calculate heading to the target (for Yaw setpoint)
+		const float drone_to_target_heading = atan2f(drone_to_target_vector(1), drone_to_target_vector(0));
+		// Actual orbit angle measured around the target, which is pointing from target to drone, so M_PI_F difference.
+		const float measured_orbit_angle = matrix::wrap_pi(drone_to_target_heading + M_PI_F);
+
+		// Update the sticks object to fetch recent data and update follow distance, angle and height via RC commands
+		_sticks.checkAndUpdateStickInputs();
+		if(_sticks.isAvailable()) {
+			updateRcAdjustedFollowHeight(_sticks);
+			updateRcAdjustedFollowDistance(_sticks, drone_to_target_vector);
+			updateRcAdjustedFollowAngle(_sticks, measured_orbit_angle, _orbit_angle_setpoint_rad);
+		}
 
-		// Set position and velocity setpoints
-		float desired_velocity_ff_scale = 0.0f;  // Used to ramp up velocity feedforward, avoiding harsh jumps in the setpoints
+		// Update target orientation to track
+		_target_course_rad = updateTargetOrientation(_target_course_rad, target_velocity_filtered.xy(), Vector3f(_follow_target_estimator.vel_est).xy());
```

Comment:
The comment is useless. Rename `updateTargetOrientation` -> `updateTargetOrientationToTrack` if you think the function name isn't clear enough (I don't think it's useful).

Reply:
Comment removed!